src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -503,8 +503,20 @@ def __init__(self, name_prefix='laworkspace', location='eastus2euap', parameter_
     def create_resource(self, name, **kwargs):
         group = self._get_resource_group(**kwargs)
         template = ('az monitor log-analytics workspace create -l {} -g {} -n {}')
-        self.live_only_execute(self.cli_ctx, template.format(self.location, group, name))
-        return {self.parameter_name: name}
+        try:
```

Comment:
May I ask why need to change codes of testsdk?

Reply:
Create containerapp managed environment need arguments `--logs-workspace-id` and `--logs-workspace-key`.
The `--logs-workspace-key` is log workspace `customerId`, ` --logs-workspace-key `is `primarySharedKey`. 
We hope it can be prepared well and the `NoTrafficRecording`, So I change the `LogAnalyticsWorkspacePreparer `to return multiple parameters include `customerId` and `primarySharedKey`.

Reply:
Since the testsdk is commonly used by every modules, it is not suggested to change the testsdk.
If it has to be changed, please open another PR.

Reply:
Hi @zhoxing-ms 
Can I update `LogAnalyticsWorkspacePreparer`  in same PR, as it has been used in test cases, which I confirm with you before.

Reply:
separate to:
https://github.com/Azure/azure-cli/pull/27314